File: Leetcode/405Hexadecimal.go
// golang 这速度也是可以了，代码也很简洁
func toHex(num int) string {
	var temp uint32 = uint32(num)
	var chars []rune = []rune("0123456789abcdef")
	var s string = ""
	for temp > 0 {
		s = string(chars[temp % 16]) + s
		temp /= 16
	}
	return s
}

File: Leetcode/405Hexadecimal.cpp
class Solution {
public:
    string toHex(int num) {
        // 为什么go版本的不需要判断0？？？
        if (num == 0)return "0";
        char chs[] = "0123456789abcdef";
        string result;
        unsigned unum = (unsigned)num;
        while(unum > 0) {
            result = chs[unum % 16] + result;
            unum /= 16;
        }

        return result;
    }
};
